content/learn/beginner/outputs-1.md

```diff
@@ -0,0 +1,85 @@
++++
+title = "Outputs 1"
+weight = 210
++++
+
+In this chapter, we will go over two ways to output text from a custom command: using the *response*, and later on using
+*template actions*.
+
+## Response
+
+If you paste text into the response field of a custom command, the bot will respond with that text when the command is
+triggered.
+
+For instance, paste the following text into a new custom command:
+
+```go
+Hello World.
+```
+
+This will make the bot respond "Hello World." in a new message when the command is triggered, as shown below.
+
+<center>
+
+![Hello World example ran in Discord.](hello_world.png)
+
+</center>
+
+## Template Actions
+
+### Actions As Output
+
+Earlier, we just made the bot respond with some static text. However, sometimes (or rather, most of the time) we want to
+change that around depending on various factors. This is where *template actions* come in.
+
+Template actions are a way to dynamically change the output depending on various things, such as the user who triggered
+the command, the arguments passed to the command, or even the current time.
+
+The bot evaluates the template action and replaces it with the result of the evaluation. For instance, the following
+code will make the bot respond with the server name when the command is triggered:
+
+```go
+{{ .Server.Name }}
+```
+
+As you can see, it differs from the static response in that it uses double curly braces (`{{` and `}}`) to denote an
+action. Without those, the bot would just respond with the text as-is.
+
+If we want to bring this a step further, we can combine the plain response with some template actions to make it a bit
+more nicer-looking:
+
+```go
+Hey there {{.User.Username}}!
+Welcome to {{ .Server.Name }}!
+```
+
+Play around with this a little bit and see what you can come up with. Take a look at the
+[data reference documentation](/docs/reference/templates/syntax-and-data) to see what other data you can access.
+
+### Actions For Computation
```

Comment:
I wonder if there is a framing of this introduction that allows a smoother transition to actions that, well, simply perform actions. For instance, `addRoleID` adds a role but is not a computation or a task per se.

Reply:
`addRoleID` is a task though—"add the role with this ID".

I do agree however, "Computation" is a bit weird, I had a mental note to change this to something better as well.

Reply:
Maybe `Actions For Functionality`? But that also reads like :sparkles: hot garbage :sparkles: